File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.js
// 2트
  // 어라.. 근데, 굳이 indexOf로 최댓값과 최댓값의 인덱스 찾을 필요 없잖아
  // 그냥 내림차순sort하고 앞에서 부터 걸러내면 될거같은데?
function solution(k, tangerine) {
  let obj = {};

  for (let i=0; i<tangerine.length; i++) {
    if (!obj[tangerine[i]]) obj[tangerine[i]] = 1
    else obj[tangerine[i]] += 1
  };

  let values = Object.values(obj); // obj에서 value값만 떼와서 모음
  values.sort((a,b) => b-a)
  
  for (let i=0; i<values.length; i++) {
    k -= values[i]
    if (k <= 0) return i+1
  }
}

console.log(solution(6,[1, 3, 2, 5, 4, 5, 2, 3,7]));
console.log(solution(4,[1, 3, 2, 5, 4, 5, 2, 3]));
console.log(solution(2,[1, 1, 1, 1, 2, 2, 2, 3]));

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
# 제한사항
    # 1 ≤ k ≤ tangerine의 길이 ≤ 100,000
    # 1 ≤ tangerine의 원소 ≤ 10,000,000
# 흠.. 먼가 과정 줄일 수 있지 않을까 싶은데..

def solution(k, tangerine):
    just_maxcheck = max(tangerine)
    dict = {x: 0 for x in range(1, just_maxcheck+1)}
    for ele in tangerine:
        dict[ele] += 1

    for_sort_list = []
    for i in range(1,just_maxcheck+1):
        for_sort_list.append([i,dict[i]])
    for_sort_list.sort(reverse=True, key=lambda x:x[1])

    types = 0
    for ele in for_sort_list:
        types += 1
        k -= ele[1]
        if k<=0:
            return types


print(solution(6,[1, 3, 2, 5, 4, 5, 2, 3]))
print(solution(4,[1, 3, 2, 5, 4, 5, 2, 3]))
print(solution(2,[1, 1, 1, 1, 2, 2, 2, 3]))